tests/integration/test_api.py

```diff
@@ -383,6 +389,39 @@ def test_annotation_event(client, monkeypatch, httpserver):
     assert result.status_code == 200
 
 
-def test_display_insight_prediction_for_products(client, mocker):
+def test_insight_prediction(client, mocker):
     result = client.simulate_get("/api/v1/insights/predictions")
     assert result.status_code == 200
+    assert result.json == {"count": 0, "predictions": [], "status": "no_predictions"}
+
+    PredictionFactory(barcode="123")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["barcode"] == "123"
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+
+    PredictionFactory(value_tag="en:seeds")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 2
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[1]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+    assert predictions_data[1]["value_tag"] == "en:seeds"
+
+
+def test_get_predictions():
+    PredictionFactory(barcode="123")
+    actual = get_predictions(barcode="123")
+    actual_items = [item.to_dict() for item in actual]
+    assert actual_items[0]["barcode"] == "123"
+    assert actual_items[0]["type"] == "category"
+    assert actual_items[0]["value_tag"] == "en:seeds"
```

Comment:
you could create at list 4 Prediction before starting with values varying for type, value_tag, barcode. Then test that using filters you get the one you want.

You can verify filter works by only verifying the ids you get.

Eg.: 
prediction1 --> barcode=123, type="category", value_tag="en:seeds"
prediciton2 --> barcode=123, type="category", value_tag="en:beers"
prediction3 --> barcode=123, type="label", value_tag="en:eu-organic"
prediction4 --> barcode=456, type="label", value_tag="en:eu-organic"

If you filter on barcode 123, you get the first 3,
if you filter on type "category" you get the first 2,
if you filter on value_tag "en:seeds", you only get the first,
if you filter on type="brand", you get none

Then you can try to combine filters like barcode=123 and type="category"